apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/db/DatabaseManagerTest.java

```diff
@@ -123,7 +130,75 @@ void createAndAddShard() throws Exception {
         TransactionalDataSource newShardDb = databaseManager.createAndAddShard(1L);
         assertNotNull(newShardDb);
         assertNotNull(newShardDb.getConnection());
+        databaseManager.shutdown(newShardDb);
+//        newShardDb.shutdown(); // not needed
+    }
+
+    @Test
+    void createShardInitTableSchemaVersion() throws Exception {
+        databaseManager = new DatabaseManager(baseDbProperties, propertiesHolder);
+        assertNotNull(databaseManager);
+        TransactionalDataSource dataSource = databaseManager.getDataSource();
+        assertNotNull(dataSource);
+        TransactionalDataSource newShardDb = databaseManager.createAndAddShard(1L, new ShardInitTableSchemaVersion());
+        assertNotNull(newShardDb);
+        assertNotNull(newShardDb.getConnection());
```

Comment:
How to verify, that shard datasource was created successfully?

Reply:
we call `newShardDb.getConnection()` for that. If we want deeper check, we should execute simple SQL on connection as do in DataSourceWrapper.java, something like: `"select X from dual;"`

Reply:
Maybe check, that all tables were create?